app/src/composables/use-page-size.ts

```diff
@@ -0,0 +1,45 @@
+import { useServerStore } from '@/stores/server';
+import { ComputedRef, computed } from 'vue';
+
+export function usePageSize<T = any>(
+	availableSizes: number[],
+	mapCallback: (value: number, index: number, array: number[]) => T,
+	fallbackSize = 25
+): { sizes: ComputedRef<T[]>; selected: number } {
+	const {
+		info: { queryLimit },
+	} = useServerStore();
+
+	const pageSizes = computed<T[]>(() => {
+		if (queryLimit === undefined || queryLimit.max === -1) {
+			return availableSizes.map(mapCallback);
+		}
+
+		const sizes = availableSizes.filter((size) => size <= queryLimit.max);
+
+		if (sizes.length === 0) {
+			sizes.push(queryLimit.max);
+		}
+
+		return sizes.map(mapCallback);
+	});
+
+	let initialSize = fallbackSize;
+
+	if (queryLimit !== undefined) {
+		initialSize = Math.min(parseLimit(queryLimit.default), parseLimit(queryLimit.max));
+	}
```

Comment:
Could we perhaps simply use the first value of `pageSizes`as the initial size?
Right now, with the default value of `100` for `QUERY_LIMIT_DEFAULT` it feels like a too high default/start value in the app. Also, the value from `QUERY_LIMIT_DEFAULT` doesn't necessarily exist in the list of page sizes,  which could be confusing.

Reply:
Hmm interesting, `100` is already the default on the API side however this gets overwritten on the App side to be `25` in most cases using the `fallbackSize`. Should we perhaps not take the `default` into account in the App? 🤔

Reply:
> Should we perhaps not take the default into account in the App? 

I think so 🤔